cni/cmd/istio-cni-taint/main.go

```diff
@@ -0,0 +1,234 @@
+		}
+	}
+}
+
+//get all critical pods and check for the readiness of pod
+//if pod is ready, check all critical pod in the node and if all of them are ready untaint the node,
+//otherwise taint it
+func podReadinessCheck(taintSetter *taint.TaintSetter) {
+	pods, err := taintSetter.ListCandidatePods()
+	if err != nil {
+		log.Fatalf("Fatal error Listing all critical pods: %+v", err)
+	}
+	for _, pod := range pods.Items {
+		if !podutils.IsPodReady(&pod) {
+			err := taintSetter.ProcessReadyPod(&pod)
+			if err != nil {
+				log.Fatalf("error: %+v in ready pod %+v in namespace %+v", err.Error(), pod.Name, pod.Namespace)
+			}
+		} else {
+			err := taintSetter.ProcessUnReadyPod(&pod)
+			if err != nil {
+				log.Fatalf("error: %+v in runeady pod %+v in namespace %+v", err.Error(), pod.Name, pod.Namespace)
+			}
+		}
+	}
+}
+func registTaints(taintSetter taint.TaintSetter) {
+	Nodes, err := taintSetter.GetAllNodes()
+	if err != nil {
+		log.Fatalf("Fatal error cannot get all pods: %+v", err)
+	}
+	for _, node := range Nodes.Items {
+		err := taintSetter.AddReadinessTaint(&node)
+		if err != nil {
+			log.Fatalf("Fatal error cannot taint node: %+v", err)
+		}
+	}
+}
+func main() {
+	loggingOptions.OutputPaths = []string{"stderr"}
+	loggingOptions.JSONEncoding = true
+	if err := log.Configure(loggingOptions); err != nil {
+		os.Exit(1)
+	}
+	options := parseFlags()
+
+	clientSet, err := clientSetup()
+	if err != nil {
+		log.Fatalf("Could not construct clientSet: %s", err)
+	}
+	taintSetter, err := taint.NewTaintSetter(clientSet, options.TaintOptions)
+	if err != nil {
+		log.Fatalf("Could not construct taint setter: %s", err)
+	}
+	logCurrentOptions(&taintSetter, options)
+	if err != nil {
+		panic(err.Error())
+	}
+	if options.RunAsDaemon {
+		id := uuid.New().String()
+		stopCh := make(chan struct{})
+		//it will be run in leader for controller configuration and running
+		run := func(ctx context.Context) {
+			tc, err := taint.NewTaintSetterController(&taintSetter)
+			if err != nil {
+				log.Fatalf("Fatal error constructing taint controller: %+v", err)
+			}
+			tc.Run(stopCh)
+		}
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		// listen for interrupts or the Linux SIGTERM signal and cancel
+		// our context, which the leader election code will observe and
+		// step down
+		ch := make(chan os.Signal, 1)
+		signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
+		go func() {
+			<-ch
+			close(stopCh)
+			log.Info("Received termination, signaling shutdown")
+			cancel()
+		}()
+		lock := &resourcelock.LeaseLock{
+			LeaseMeta: metav1.ObjectMeta{
+				Name:      LeassLockName,
+				Namespace: LeaseLockNamespace,
+			},
+			Client: clientSet.CoordinationV1(),
+			LockConfig: resourcelock.ResourceLockConfig{
+				Identity: id,
+			},
+		}
+		leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
+			Lock:            lock,
+			ReleaseOnCancel: true,
+			LeaseDuration:   3 * time.Second,
```

Comment:
This is too fast I think, you are likely to impact api-server with this many writes